gotoalert/events.py

```diff
@@ -431,3 +445,99 @@ def get_skymap(self, nside=128):
+
+        # Default params
+        self.notice = EVENT_DICTONARY[self.packet_type]['notice_type']
+        self.type = EVENT_DICTONARY[self.packet_type]['event_type']
+        self.source = EVENT_DICTONARY[self.packet_type]['source']
+
+        # Get the run and event ID (e.g. 13311922683750)
+        self.id = top_params['AMON_ID']['value']
+
+        # Create our own event name (e.g. ICECUBE_GOLD_EVENT_13311922683750)
+        self.name = '{}_{}_{}'.format(self.source, self.notice, self.id)
+
+        # Get info from the VOEvent
+        # signalness: the probability this is an astrophysical signal relative to backgrounds
+        self.signalness = float(top_params['signalness']['value'])
+        self.far = float(top_params['FAR']['value'])
+
+        # Position coordinates
+        self.position = vp.get_event_position(self.voevent)
+        self.coord = SkyCoord(ra=self.position.ra, dec=self.position.dec, unit=self.position.units)
+        self.target = FixedTarget(self.coord)
+
+        # Position error
+        self.coord_error = Angle(self.position.err, unit=self.position.units)
+        
+        # Systematic error for cascade event is given, so = 0
+        if self.notice == 'ICECUBE_CASCADE':
+            self.systematic_error = Angle(0, unit='deg')
+        else:
+            self.systematic_error = Angle(.2, unit='deg')
+        self.total_error = Angle(np.sqrt(self.coord_error ** 2 + self.systematic_error ** 2),
+                                 unit='deg')
+
+        # Enclosed skymap url for CASCADE_EVENT, but others
+        # Get skymap URL
+        for group in group_params:
+            if 'skymap_fits' in group_params[group]:
+                self.skymap_url = group_params[group]['skymap_fits']['value']
+                self.skymap_type = group
+            else:
+                self.skymap_url = None
+
+        # Don't download the skymap here, it may well be very large.
+        # Only do it when it's absolutely necessary
+        # These params will only be set once the skymap is downloaded
+        self.contour_areas = {0.5: None, 0.9: None}
+
+    def get_skymap(self, nside=128):
+        """Download the Event skymap and return it as a `gototile.skymap.SkyMap object."""
+        if self.skymap:
+            return self.skymap
+
+        # Download the skymap from the URL
+        # The file gets stored in /tmp/
+        # Don't cache, force redownload every time
+        # https://github.com/GOTO-OBS/goto-alert/issues/36
+        if self.skymap_url:
+            self.skymap_file = download_file(self.skymap_url, cache=False)
+            self.skymap = SkyMap.from_fits(self.skymap_file)
+            self.skymap.regrade(nside)
+        else:
+            self.skymap = SkyMap.from_position(self.coord.ra.deg,
+                                               self.coord.dec.deg,
+                                               self.total_error.deg,
+                                               nside=nside)
```

Comment:
Right now this fails, because of an error from GOTO-tile. I will push an update to fix that, for now it's best to fall back to the positions in the notice (just like we do for Fermi events).

```suggestion
        if self.skymap_url:
            self.skymap_file = download_file(self.skymap_url, cache=False)
            self.skymap = SkyMap.from_fits(self.skymap_file)
            self.skymap.regrade(nside)
        except Exception:
                # Fall back to creating our own
                pass

        # Create a Gaussian skymap (if we didn't download one above)
        if not self.skymap:
            self.skymap = SkyMap.from_position(self.coord.ra.deg,
                                               self.coord.dec.deg,
                                               self.total_error.deg,
                                               nside=nside)
```